post_office/migrations/0010_message_id.py

```diff
@@ -0,0 +1,39 @@
+import random
+from django.db import migrations, models
+
+from post_office.models import STATUS
+from post_office.settings import get_message_id_enabled, get_message_id_fqdn
+
+
+def forwards(apps, schema_editor):
+    if not get_message_id_enabled():
+        return
+    msg_id_fqdn = get_message_id_fqdn()
+    Email = apps.get_model('post_office', 'Email')
+    for email in Email.objects.using(schema_editor.connection.alias).filter(message_id__isnull=True):
+        if email.status in [STATUS.queued, STATUS.requeued]:
+            # create a unique Message-ID for all emails which have not been send yet
+            randint1, randint2 = random.getrandbits(64), random.getrandbits(16)
+            email.message_id = '<{}.{}.{}@{}>'.format(email.id, randint1, randint2, msg_id_fqdn)
+            email.save()
```

Comment:
I don't think we should need to provide a forward migration to backfill old emails.

Reply:
This migration only applies if the `MESSAGE_ID_ENABLED` settings is `True` **and only** for unsent emails. I believe, most users would expect this behaviour.